console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,173 @@
+	// +kubebuilder:validation:Optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorize indicates if the proxied request should contain the logged-in user's
+	// OpenShift access token in the "Authorization" request header. For example:
+	//
+	// Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0
+	//
+	// By default the access token is not part of the proxied request.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=false
+	Authorize bool `json:"authorize"`
+}
+
+// ProxyType is an enumeration of available proxy types
+// +kubebuilder:validation:Pattern=`^(Service)$`
+type ConsolePluginProxyType string
+
+const (
+	// ProxyTypeService is used when proxying communication to a Service
+	ProxyTypeService ConsolePluginProxyType = "Service"
+)
+
+// ProxyTypeServiceConfig holds information on Service to which
+// console's backend will proxy the plugin's requests.
+type ConsolePluginProxyServiceConfig struct {
+	// name of Service that the plugin needs to connect to.
+	// The name should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Name string `json:"name"`
+	// namespace of Service that the plugin needs to connect to.
+	// The namespace should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Namespace string `json:"namespace"`
+	// port on which the Service that the plugin needs to connect to
+	// is listening on.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=65535
+	// +kubebuilder:validation:Minimum:=1
+	Port int32 `json:"port"`
+}
+
+// ConsolePluginService holds information on Service that is serving
+// console dynamic plugin assets.
+type ConsolePluginService struct {
+	// name of Service that is serving the plugin assets.
+	// The name should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Name string `json:"name"`
+	// namespace of Service that is serving the plugin assets.
+	// The namespace should be between 1 and 128 characters.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	Namespace string `json:"namespace"`
+	// port on which the Service that is serving the plugin is listening to.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=65535
+	// +kubebuilder:validation:Minimum:=1
+	Port int32 `json:"port"`
+	// basePath is the path to the plugin's assets. The primary asset it the
+	// manifest file called `plugin-manifest.json`, which is a JSON document
+	// that contains metadata about the plugin and the extensions.
+	// Defaults to the root path.
+	// +kubebuilder:validation:Required
```

Comment:
Fields must be optional if they are to be defaulted. Required vs optional is processed before defaulting.

Reply:
This comment still needs to be addressed